core/admin_menu/class-btm-admin-task-page-table.php

```diff
@@ -70,68 +70,77 @@ public function on_hook_page_render_tasks(){
 	}
 
 	public function on_hook_page_load_process_bulk(){
-		// Bulk Action Delete
-		if ( ! empty( $_GET['action'] ) && static::BULK_ACTION_DELETE === $_GET['action'] ) {
-			// todo: bulk actions should be done with POST request and nonce should be checked
-			$to_delete = $_GET[ 'record' ];
-			if( ! is_array( $to_delete ) ){
-				$to_delete = array( $to_delete );
-			}
 
-			$deleted = BTM_Task_Dao::get_instance()->delete_many_by_ids( $to_delete );
-			// todo: check $deleted, show admin notice success or error
+        if( ! empty( $_GET[ 'action' ] ) ){
+	        // Bulk Action Delete
+	        if ( static::BULK_ACTION_DELETE === $_GET['action'] ) {
+		            $this->delete_tasks();
+	        } else {
+		            $this->slash_url();
+	        }
+	        // Bulk Action Pause
+	        if ( static::BULK_ACTION_PAUSE === $_GET['action'] ) {
+		        $this->pause_tasks();
+	        } else {
+		        $this->slash_url();
+	        }
+	        // Bulk Action Resume
+	        if ( static::BULK_ACTION_RESUME === $_GET['action'] ) {
+		        $this->resume_tasks();
+	        } else {
+		        $this->slash_url();
+	        }
+        }
+	}
 
-			if ( ! empty( $_REQUEST['_wp_http_referer'] ) ) {
-				wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce', 'action', static::BULK_ACTION_DELETE, ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
-				exit;
-			}
-		} else {
-			if ( ! empty( $_REQUEST['_wp_http_referer'] ) ) {
-				wp_redirect( remove_query_arg( array( '_wp_http_referer', '_wpnonce' ), wp_unslash( $_SERVER['REQUEST_URI'] ) ) );
-				exit;
-			}
+	private function slash_url(){
```

Comment:
@eriktad why we need this method ?

Reply:
I remove this method... Now we don't need this functionality